### PR TITLE
Add support for multiple games including pre-written levels.

### LINKED ISSUE #79: Make a `Game` model, use it in the API

### PR DESCRIPTION
**Pull request by joshuablake**
- Refactor models so that each Game is stored and users can have multiple Avatars (one per game)
- aimmo-game-creator now has a WorkerManger (similar to aimmo-game) that can control multiple concurrent games.
- Can add pre-written level 'generators' to aimmo-game for deterministic outputs.
- Configuration for each game is now stored in the DB.

Base-work for #21 (no undo support). Fixes #79.

### CODE REVIEW

File: aimmo-game-creator/service.py [hunk 1]
```diff
@@ -8,9 +8,11 @@
 
 LOGGER = logging.getLogger(__name__)
 
+
 def create_game_rc(api, name, environment_variables):
     environment_variables['SOCKETIO_RESOURCE'] = "game/%s/socket.io" % name
-    environment_variables['GAME_API_URL'] = 'https://staging-dot-decent-digit-629.appspot.com/aimmo/api/games/'
+    environment_variables['GAME_API_URL'] =\
+        'https://staging-dot-decent-digit-629.appspot.com/aimmo/api/games/'
```
Comment (mikebryant) [inline aimmo-game-creator/service.py]: I'm really not a fan of this syntax. Also, we don't care about the 80 character line limits in pep8

File: players/views.py [hunk 2]
```diff
@@ -60,10 +70,23 @@ def get_game(request, id):
...
+
+
 class WatchView(TemplateView):
     template_name = 'players/watch.html'
 
     def get_context_data(self, **kwargs):
         context = super(WatchView, self).get_context_data(**kwargs)
-        context['game_url_base'], context['game_url_path'] = app_settings.GAME_SERVER_LOCATION_FUNCTION('main')
+        context['game_url_base'], context['game_url_path'] =\
+            app_settings.GAME_SERVER_LOCATION_FUNCTION(kwargs['id'])
```
Comment (mikebryant) [inline players/views.py]: Again, I don't think we want to be using this style

**Comment by joshuablake** [comment]
Now fixed so works with new run script.

**Comment by joshuablake** [comment]
Think this is now complete, will edit main post to better explain the changes.

**Comment by joshuablake** [comment]
This is now working under the minikube config in #135 (with a tweak to test.sh you can see in 5da34e).

**Comment by Spycho** [comment]
Merged and slightly (but not fully) working code pushed to https://github.com/ocadotechnology/aimmo/tree/joshuablake-multiple_games. Outstanding issues:
- aimmo-game tests fail with a very odd seemingly import related error that goes away if you comment out some of the imports in maps.py. Fairly useless stack trace.
- it looks like score locations and pickups are not spawning
- on creating a new game, it didn't look like the avatar spawned properly. Restarting the server seemed to resolve. I may have just not been patient enough. Further testing required.
- "levels" don't display

**Pull request merged**